### ISSUE TITLE
Exception in thread "main" java.lang.NoClassDefFoundError: javafx/application/Application$Parameters

### ISSUE THREAD

**Issue by blshkv** [Col-E/Recaf#123]
This is a continuation of the following:
https://github.com/Col-E/Recaf/issues/72
https://github.com/Col-E/Recaf/issues/79

I have exactly the same issue and tried tree different JDK:
```
  [1]   icedtea-bin-8  system-vm
  [2]   openjdk-bin-11 
  [3]   oracle-jdk-bin-11  user-vm
```
Here is the last error:
```
/opt/oracle-jdk-bin-11.0.2/bin/java -jar recaf-1.13.1.jar 
Picked up _JAVA_OPTIONS: -Dawt.useSystemAAFontSettings=on
Exception in thread "main" java.lang.NoClassDefFoundError: javafx/application/Application$Parameters
        at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
        at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3167)
        at java.base/java.lang.Class.getDeclaredMethods(Class.java:2310)
        at me.coley.event.AccessHelper.getMethodsRecursively(AccessHelper.java:347)
        at me.coley.event.EventBus.getInvokers(EventBus.java:128)
        at me.coley.event.EventBus.subscribe(EventBus.java:51)
        at me.coley.event.EventBus.subscribe(EventBus.java:75)
        at me.coley.event.Bus.subscribe(Bus.java:46)
        at me.coley.recaf.Recaf.main(Recaf.java:20)
Caused by: java.lang.ClassNotFoundException: javafx.application.Application$Parameters
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        ... 9 more

````

After some googling, I have discovered the following:
https://www.infoworld.com/article/3305073/removed-from-jdk-11-javafx-11-arrives-as-a-standalone-module.html

Could you add a standalone module as suggested or re-write a  tool without that library?

**Comment by blshkv** [comment]
As PoC, I have downloaded https://gluonhq.com/products/javafx/ as suggested in the link above, unpacked it into "javafx" directory and managed to run the tool with the following:
```
java -cp 'recaf-1.13.1.jar:javafx/*' me.coley.recaf.Recaf
```

**Comment by Col-E** [comment]
I've not set my sights on cross-version _(Because java 8 to java 9+ is more complex than prior major version updates)_ compatibility yet. As far as I'm aware Java 8 is still the most widely installed version, so I'm targeting it. The extent of my Java 9+ support is preventing crashes like this, while pointing users to a quick update guide: https://col-e.github.io/Recaf/jigsaw.html

I have removed the usage of `Application$Parameters` 

Regarding the _"config"_ menu not working, the ControlsFx dependency will need to be updated as well as mentioned in the update guide.

**Comment by blshkv** [comment]
regarding the update guide, you should do it automatically using profiles:
https://stackoverflow.com/questions/36043605/pass-jvm-args-to-maven-surefire-conditional-on-build-java-version
https://maven.apache.org/guides/introduction/introduction-to-profiles.html

**Comment by Col-E** [comment]
Fair point, I had tried using profiles before and couldn't get them to work. Now they seem to be fine. I'll make a commit for this suggestion.

Regarding my removal of `Application$Parameters`, sometimes there are cases where I'd need two version-specific source files. Those are the cases I'm trying to eliminate.

**Comment by Col-E** [comment]
Closing since this specific problem has been resolved as of 6a48b20ade51978719256c574b0dd966146d925c

Another thread on better cross version compatibility in general is warranted though.
